modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -2670,6 +2738,21 @@ private boolean readPageFromStore(long pageId, ByteBuffer buff) throws IgniteChe
 
             assert read : toDetailString(pageId);
 
+            if (PageIO.getCompressionType(buff) != CompressionProcessor.UNCOMPRESSED_PAGE) {
+                int comprPageSz = PageIO.getCompressedSize(buff);
+
+                compressProc.decompressPage(buff, store.getPageSize());
+
+                if (punchHoleEnabled && comprPageSz < store.getPageSize()) {
+                    try {
+                        store.punchHole(pageId, comprPageSz);
+                    }
+                    catch (Exception ignored) {
```

Comment:
Why we should ignore it?

Reply:
Because there is no need to fail here, punching holes is optional.